### PR TITLE
Enable glean network logging and prepare for burnham 21.1.0 release.

### PR DESCRIPTION
**Pull request by jrbenny35**
Works on #351

### CODE REVIEW

**Review by Dexterp37 — APPROVED** [review]

File: application/setup.py [hunk 1]
```diff
@@ -13,7 +13,7 @@ def read(*args: str) -> str:
 
 setuptools.setup(
     name="burnham",
-    version="21.0.0",
+    version="21.1.0",
```
Comment (hackebrot) [inline application/setup.py:16]:
```suggestion
    version="22.0.0",
```

File: application/src/burnham/__init__.py [hunk 2]
```diff
@@ -2,7 +2,7 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-__version__ = "21.0.0"
+__version__ = "21.1.0"
```
Comment (hackebrot) [inline application/src/burnham/__init__.py:5]:
```suggestion
__version__ = "22.0.0"
```

File: application/src/burnham/cli.py [hunk 3]
```diff
@@ -127,6 +127,7 @@ def burnham(
         upload_enabled=enable_telemetry is True,
         data_dir=Path(TemporaryDirectory().name),
         configuration=Configuration(server_endpoint=platform),
+        log_level=logging.DEBUG,
```
Comment (hackebrot) [inline application/src/burnham/cli.py:130]: Before adding this new logging capability to Glean, we would control the log level for Glean on the burnham application level by setting `logging.basicConfig(level=logging.DEBUG)` in the lines above this one.

My recommendation would be to **not** hardcode the Glean `log_level`, but use the same logic as for the burnham application and set the `log_level` based on the `--verbose` CLI flag.

The burnham DAG in telemetry-airflow sets this to `true`.

https://github.com/mozilla/telemetry-airflow/blob/main/dags/burnham.py#L427

**Review by hackebrot — CHANGES_REQUESTED** [review]
Hi @jrbenny35! 👋🏻 

I added one inline comment about logging. Please note that since we use Calendar Versioning the version identifier for the next release would be `22.0.0`.

File: application/src/burnham/__init__.py (hunk 2 above)
Comment (jrbenny35) [reply application/src/burnham/__init__.py:5]: Thought it was semver

Comment (hackebrot) [reply application/src/burnham/__init__.py]: No, we wouldn't be at a major release `21` today.

The versioning scheme is briefly mentioned on the README at https://github.com/mozilla/burnham/tree/main/application#installation

Comment (jrbenny35) [reply application/src/burnham/__init__.py]: heh, that would be a lot of releases.

**Review by hackebrot — APPROVED** [review]
LGTM! 🚀

**Pull request merged**